great_expectations/validator/v1_validator.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import TYPE_CHECKING
+
+from great_expectations.core.expectation_validation_result import (
+    ExpectationSuiteValidationResult,
+    ExpectationValidationResult,
+)
+from great_expectations.validator.validator import calc_validation_statistics
+
+if TYPE_CHECKING:
+    from great_expectations.core import ExpectationSuite
+    from great_expectations.core.batch_config import BatchConfig
+    from great_expectations.data_context import AbstractDataContext
+    from great_expectations.expectations.expectation import (
+        Expectation,
+        ExpectationConfiguration,
+    )
+
+
+class ResultFormat(str, Enum):
+    BOOLEAN_ONLY = "BOOLEAN_ONLY"
+    BASIC = "BASIC"
+    SUMMARY = "SUMMARY"
+    COMPLETE = "COMPLETE"
+
+
+class Validator:
+    """Validator.
+
+    Responsible for running expectations on a batch configuration.
+    """
+
+    _context: AbstractDataContext
+    _batch_config: BatchConfig
+    _batch_asset_options: dict | None
+    result_format: ResultFormat
+
+    def __init__(
+        self,
+        context: AbstractDataContext,
+        batch_config: BatchConfig,
+        result_format: ResultFormat = ResultFormat.SUMMARY,
+        batch_asset_options: dict | None = None,
+    ) -> None:
+        self._context = context
+        self._batch_config = batch_config
+        self._batch_asset_options = batch_asset_options
+        self.result_format = result_format
```

Comment:
Could we also make this private? If we want to expose it we can do so with a `property`?

Reply:
I'm fine either way, but what does a property get us that an attribute doesn't?

Reply:
we can make it read-only - essentially just introducing getters/setters to limit access if that's what we desire

Reply:
This is something the user may want to change while interactively playing with the Validator so we want it to be mutable. We could leave it public now and change to a property later if we want.